### PR TITLE
Config for cookie expire, logDirectory and jsUrl

### PR DESCRIPTION
**Pull request by bstoney**
- Added `expire` to `cookieConfig` to allow setting the cookie expire time in seconds from now
- Changed `logDirectory` to allow absolute paths for the log files 
- Changed `jsUrl` to use `FALSE` to disable injection of the CSRF guard JavaScript and allow it to be added to the site manually

Also updated some of the doc comments to correct usage and spelling.

### CODE REVIEW

**Comment by mebjas** [comment]
@bstoney PR looks good, thanks for that; 

I have completed my review, there are some comments that you need to look at - please correct them and I'll merge;

**Comment by bstoney** [comment]
Cool, thanks @mebjas , where do I find your comments?

**Comment by bstoney** [comment]
@mebjas any chance you can look at this PR? I am happy to make any additional changes if necessary. Thanks

**Comment by mebjas** [comment]
Apologies for the delays. Are the review comments not available to you? https://github.com/mebjas/CSRF-Protector-PHP/pull/97/files

**Comment by bstoney** [comment]
Thanks @mebjas the comments are available to me now

File: libs/csrf/csrfprotector.php [hunk 1]
```diff
@@ -164,7 +165,7 @@ public static function init($length = null, $action = null)
 
 			//start session in case its not
 			if (session_id() == '')
-			    session_start();
+				session_start();
```
[the remark this answers is not in the document]
Comment (bstoney) [reply libs/csrf/csrfprotector.php]: This line is indented with a single tab, which matches the rest of the file

File: libs/csrf/csrfprotector.php [hunk 2]
```diff
@@ -382,6 +381,8 @@ private static function failedValidationAction()
 					//redirect to custom error page
 					$location  = self::$config['errorRedirectionPage'];
 					header("location: $location");
+					exit(self::$config['customErrorMessage']);
```
[the remark this answers is not in the document]
Comment (bstoney) [reply libs/csrf/csrfprotector.php]: This case statement used to fall through to the next case. As this is not usually considered good practice I added the break statement and needed to duplicate the exit message from the next case. I admit repeated code it also not good practice, but I believe it it easier to follow than a case fall through.

File: test/config.testInit_incompleteConfigurationException.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * ---- failedAuthAction
+ * ---- jsUrl
+ * ---- tokenLength
+ */
+return array(
+	"CSRFP_TOKEN" => "csrfp_token",
+//	"logDirectory" => "../log",
+//	"failedAuthAction" => array(
+//		"GET" => 0,
+//		"POST" => 0),
```
[the remark this answers is not in the document]
Comment (bstoney) [reply test/config.testInit_incompleteConfigurationException.php]: I added a number of test which use these config files and the new functionality, see testInit_incompleteConfigurationException, testCookieExpireTime, and others

File: test/config.testInit_withoutInjectedCSRFGuardScript.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Configuration file for CSRF Protector
+ * Necessary configurations are (library would throw exception otherwise)
+ * ---- logDirectory
+ * ---- failedAuthAction
+ * ---- jsUrl
+ * ---- tokenLength
+ */
+return array(
+	"CSRFP_TOKEN" => "csrfp_token",
```
[the remark this answers is not in the document]
Comment (bstoney) [reply test/config.testInit_withoutInjectedCSRFGuardScript.php]: As above

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,10 @@
 phpunit.phar
-coverage/*
\ No newline at end of file
+coveralls.phar
+
+coverage/
+vendor/
+build/
+log/*.log
```
[the remark this answers is not in the document]
Comment (bstoney) [reply .gitignore]: Done

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,10 @@
...
+
+# PHPStorm project directory
+.idea/*
```
[the remark this answers is not in the document]
Comment (bstoney) [reply .gitignore]: Done

File: libs/csrf/csrfprotector.php [hunk 7]
```diff
@@ -496,45 +497,50 @@ public static function generateAuthToken()
...
+					return $buffer;
+				}
+			}
+
+			// TODO: statically rewrite all forms as well so that if a form is submitted
+			// before the js has worked on, it will still have token to send
+			// @priority: medium @labels: important @assign: mebjas
+			// @deadline: 1 week
+
+			if (self::$config['jsUrl']) {
```
[the remark this answers is not in the document]
Comment (bstoney) [reply libs/csrf/csrfprotector.php]: Reverted

File: .gitignore (hunk 5 above)
[the remark this answers is not in the document]
Comment (mebjas) [reply .gitignore]: Please push the latest code.

File: libs/csrf/csrfprotector.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (mebjas) [reply libs/csrf/csrfprotector.php]: Ok yes.

File: libs/csrf/csrfprotector.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (mebjas) [reply libs/csrf/csrfprotector.php]: Ok seems good.

**Comment by mebjas** [comment]
@bstoney Hi most of the things look good except for `.gitignore`; 

And you know what you did some great work on this PR; Lot of this recently came as ask from some users.

Highly appreciate your contribution!

**Comment by mebjas** [comment]
Actually it's fine; Merging your pull request!

**Comment by bstoney** [comment]
Cool thanks @mebjas

**Pull request merged**